File: 2° Periodo/AEDS 2/Exercícios avulsos/beecrowd/organizadorVag.java
import java.util.*;

public class organizadorVag {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine());
        
        while(n!=0) {
            int L = sc.nextInt();
            sc.nextLine();
            String vagoes = sc.nextLine();
            String[] posicao = vagoes.split(" ");
            int trocas = 0;

            for(int i=0; i<L-1; i++) {
                //int x = Integer.parseInt(posicao[i]); //trocando para inteiros para conseguir fazer a comparacao
                for(int j=0; j<L-1-i; j++) {
                    //int y = Integer.parseInt(posicao[j]);

                    if(Integer.valueOf(posicao[j]) > Integer.valueOf(posicao[j+1])) {
                        //swap
                        String tmp = posicao[j];
                        posicao[j] = posicao[j+1];
                        posicao[j+1] = tmp;

                        trocas++;
                    }
                }
            }

            System.out.println("Optimal train swapping takes " + trocas + " swaps.");
            n--;
        }
    }
}

File: 2° Periodo/AEDS 2/Exercícios avulsos/beecrowd/organizadorVag.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int n = 0;
    scanf("%d", &n);

    while(n!=0) {
        int tam = 0, trocas;
        scanf("%d", &tam);

        //alocando espaco para o array de vagoes
        int *vagoes = (int*)malloc(tam * sizeof(int));
        for(int i=0; i<tam; i++) {
            scanf("%d", &vagoes[i]);
        }

        for(int i=0; i<tam; i++) {
            for(int j=i+1; j<tam; j++) {
                if(vagoes[i] > vagoes[j]) {
                    trocas++; //n há necessidade de fazer o swap pq n é o que o exercicio pede, apenas saber que há uma troca sendo feita
                }
            }
        }

        printf("Optimal train swapping takes %d swaps.\n", trocas);
        n--;
    }
    return 0;
}